wezterm-gui/src/termwindow/mod.rs

```diff
@@ -1756,17 +1757,17 @@ impl TermWindow {
         Ok(())
     }
 
-    fn scroll_by_page(&mut self, amount: isize) -> anyhow::Result<()> {
+    fn scroll_by_page(&mut self, amount: NotNan<f64>) -> anyhow::Result<()> {
         let pane = match self.get_active_pane_or_overlay() {
             Some(pane) => pane,
             None => return Ok(()),
         };
         let dims = pane.get_dimensions();
         let position = self
             .get_viewport(pane.pane_id())
-            .unwrap_or(dims.physical_top)
-            .saturating_add(amount * dims.viewport_rows as isize);
-        self.set_viewport(pane.pane_id(), Some(position), dims);
+            .unwrap_or(dims.physical_top) as f64
+            + (f64::from(amount) * dims.viewport_rows as f64);
```

Comment:
I think you can deref the `NotNan` to get at the inner value; can you try this?
I think this is slightly easier to read:

```suggestion
            + (*amount * dims.viewport_rows as f64);
```